source/_posts/angular-signals.md

```diff
@@ -0,0 +1,409 @@
+  `,
+})
+export class AppComponent implements OnInit {
+   counter = signal(0);
+   isEqual = computed(() => this.counter() === 5);
+   isOdd = computed(() => this.counter() % 2);
+
+   ngOnInit() {
+      effect(() => {
+         console.log('Counter value', this.counter());
+
+         if (this.counter() === 5) {
+            console.log('Counter value is equal 5!');
+         }
+      });
+   }
+
+   increment() {
+      this.counter.update((value) => value + 1);
+   }
+
+   decrement() {
+      this.counter.update((value) => value - 1);
+   }
+
+   reset() {
+      this.counter.set(0);
+   }
+}
+```
+
+## Zoneless Angular
+
+By default, Angular change detection is run globally, where the entire component tree is checked from top to bottom. This is made possible by the Zone.js library, which captures all async events such as:
+
+* `setTimeout`,
+* event listeners,
+* requests,
+* and other. 
+
+How does this look in practice?
+
+```ts
+@Component({
+   selector: 'app-root',
+   template: 'Welcome {{name}}!'
+})
+export class AppComponent implements OnInit {
+   name!: string;
+
+   ngOnInit() {
+      setTimeout(() => {
+         this.name = 'Lucas';
+      }, 2000);
+   }
+}
+```
+
+1. Zone.js detects `setTimeout`.
+2. It gives a signal to run change detection automatically.
+3. Lastly, after two seconds we receive the result `Welcome Lucas!`.
+
+{% add_classes "Default change detection strategy" "center bold"%}
+
+{% img "change-detection-strategy-default.png" "Default change detection" "lazy" %}
+
+Automatic change detection can lead to many problems if we write our application the wrong way, including impact on the performance. Certain strategies and methodologies are usually used for this to create more performant and responsive applications. What if we wanted Angular to be able to run without Zone.js so that it would not automatically update the application for us?
+
+### Angular without Zone.js
+
+We can completely remove Zone.js from our application. To do this, simply deactivate Zone.js and set `ngZone` to `noop` when bootstrapping the app.
+
+```ts main.ts
+import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+import { AppModule } from './app/app.module';
+
+platformBrowserDynamic()
+        .bootstrapModule(AppModule, { ngZone: 'noop' })
+        .catch((err) => console.error(err));
+```
+
+At this point you must trigger change detection on your own. This requires comprehensive knowledge of change detection. If we would like to render our `name` now, we can trigger the change detection manually with `tick` or `detectchanges()`.
+
+```ts
+constructor(private cdr: ChangeDetectorRef,
+            private applicationRef: ApplicationRef) {}
+
+ngOnInit() {
+   setTimeout(() => {
+      this.name = 'Lucas';
+      this.cdr.detectChanges(); // runs change detection for the underlying component
+      // this.applicationRef.tick(); // runs application-wide change detection
```

Comment:
Why comment?

Reply:
This is an example, you can use detectCHanges to trigger changes or tick. I wanted to show to the user the possibility.

Reply:
But this looks like a mistake - maybe find a better way to present your idea?